src/main/java/com/github/curriculeon/SimplePhoneBook.java

```diff
@@ -1,32 +1,62 @@
 package com.github.curriculeon;
 
+import java.util.ArrayList;
+import java.util.HashSet;
 import java.util.List;
+import java.util.Set;
+import java.util.concurrent.LinkedTransferQueue;
 
 /**
  * Created by leon on 12/16/2019.
  */
 public class SimplePhoneBook {

```

Comment:
I would recommend creating a field of type `AdvancedPhoneBook` in this class.
Then, use defer all method calls to `advancedPhoneBook` instance.